s3/src/main/scala/geotrellis/spark/io/s3/S3GeoTiffRDD.scala

```diff
@@ -120,40 +120,51 @@ object S3GeoTiffRDD extends LazyLogging {
   /**
     * Creates a RDD[(K, V)] whose K and V  on the type of the GeoTiff that is going to be read in.
     *
-    * @param bucket   Name of the bucket on S3 where the files are kept.
-    * @param prefix   Prefix of all of the keys on S3 that are to be read in.
-    * @param uriToKey function to transform input key basing on the URI information.
-    * @param options  An instance of [[Options]] that contains any user defined or default settings.
+    * @param  bucket    Name of the bucket on S3 where the files are kept.
+    * @param  prefix    Prefix of all of the keys on S3 that are to be read in.
+    * @param  uriToKey  Function to transform input key basing on the URI information.
+    * @param  options   An instance of [[Options]] that contains any user defined or default settings.
+    * @param  geometry  An optional geometry to filter by.  If this is provided, it is assumed that all GeoTiffs are in the same CRS, and that this geometry is in that CRS.
     */
-  def apply[I, K, V](bucket: String, prefix: String, uriToKey: (URI, I) => K, options: Options)
-    (implicit sc: SparkContext, rr: RasterReader[Options, (I, V)]): RDD[(K, V)] = {
+  def apply[I, K, V](
+    bucket: String, prefix: String,
+    uriToKey: (URI, I) => K,
+    options: Options,
+    geometry: Option[Geometry] = None
```

Comment:
Hm, mb to add some helpful overloads? Though it's smth to discuss, as the best otpion would be to move it into `Options` and it's impossible due to compatibility reasons. ):

Reply:
There also needs to be an `apply` with the old signature to avoid removing this method from the API.